### ISSUE TITLE
Tacotron GST: Inference not working

### ISSUE THREAD

**Issue by ErnstTmp** [NVIDIA/OpenSeq2Seq#302]
I am running Tacotron GST with mailabs or LJ.

In Audio tab in Tensorboard, the Train_audio sounds good after the model has converged. 

However, if I feed an audio from the training set for inference, the generated samples are not understandable (even if I feed the same audio & text as in the Tensorboard)

[generate.csv.zip](https://github.com/NVIDIA/OpenSeq2Seq/files/2648307/generate.csv.zip)

I run inference with 
<pre>
python run.py --config_file=example_configs/text2speech/tacotron_gst_enu.py  --mode=infer --infer_output_file=unused
</pre>
Thanks a lot, Ernst

**Comment by blisc** [comment]
Can you provide a sample of the eval audio from training and a sample of the learned attention alignments?

Does the problem still persist with the provided trained checkpoint? (Note it was trained with MAILABS data)

Train audio is not a good indication of model convergence. A good way to tell is if the eval audio is good or the learned attention alignments (train and/or eval) are approximately diagonal.

**Comment by ErnstTmp** [comment]
1) If I use the trained checkpoint, all works well
2) The eval audio sounds bad
3) The eval loss is bad, see enclosed file. Training loss looks good.
![eval](https://user-images.githubusercontent.com/17543131/49599535-8e66cf80-f981-11e8-97d2-abc00b0a5540.png)
4) The alignments are not diagonal at all
5) All training / Eval Data is from mailabs German data set

Can you give me an advice how to proceed?

**Comment by blisc** [comment]
I never tested tacotron GST on the German MAILABS data set so it is possible that tacotron GST is unable to learn that dataset.

I would advise you to limit the dataset to just one speaker and see if the eval audio becomes recognizable.
If that works, try adding an additional speaker one at a time. If that doesn't work, then try further limiting it to just one book.

I would also advise you to play around with the emb_size, attention_layer_size, num_tokens, and num_heads parameters for the Style Encoder.

If you get it working, please let us know!

**Comment by blisc** [comment]
For German, I have seen success using "Toten Seelen" read by Eva K using the regular tacotron model without GST.

For both German and en_US, I trained the models on 1 GV100 with a batch size of 32. I would recommend using 32 files for eval.

**Comment by ErnstTmp** [comment]
Hi blisc, thanks a lot! The batchsize seem to be the problem. I trained on Titan X, and the memory was too small, so I had to reduce batch size. I will retry on V 100 with batch size 32.
